cnd/src/swap_protocols/ledger/bitcoin.rs

```diff
@@ -1,56 +1,50 @@
 use crate::comit_api::LedgerKind;
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub struct Mainnet;
-
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub struct Testnet;
-
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
-pub struct Regtest;
+macro_rules! declare_ledger {
```

Comment:
I vote no on this one. The result is longer than the original code which means to me that it is not beneficial. Or would you want to re-use it for Ethereum?

Reply:
Original total lines of code for this file is 56 lines, macro version is 50.

I would not re-use these macros for the ethereum side just because it makes the code harder to read, easier to have the macro code write were its called I feel.  Also because the macros rely on identifiers in the file its better to have the macro in the file also.